spring-integration-core/src/main/java/org/springframework/integration/handler/AbstractMessageProducingHandler.java

```diff
@@ -139,7 +139,7 @@ public void setNotPropagatedHeaders(String... headers) {
 	 * @param merge true to merge with existing patterns; false to replace.
 	 * @since 5.0.2
 	 */
-	protected final void updateNotPropagatedHeaders(String[] headers, boolean merge) {
+	protected void updateNotPropagatedHeaders(String[] headers, boolean merge) {
```

Comment:
Well, I see now JavaDocs. Not sure why I have missed to read them :cry: 

The problem that this method nowhere is used from the ctor...

What am I missing? Maybe we should just relax this method and let end-user to deal with it on its own?

I think otherwise I will override the setter and adder in the `MessageTransformingHandler`.

Thanks

Reply:
Wait, but those headers are not mapped by the `AbstractKafkaHeaderMapper` anyway.
Don't we use there a `headerMapper` for gateway reply messages?

Reply:
It has nothing to do with header mapping either outbound or reply; those headers might be present on the outbound message received by the gateway, and we don't want them to appear in the reply message. Hence they are added to the `notPropagatedHeaders`.